econml/utilities.py

```diff
@@ -1303,6 +1304,48 @@ def predict(self, X):
         return self._clf.predict_proba(X)[:, 1:]
 
 
+def _shap_explain_cme(cme_model, X, d_y, d_t, feature_names=None, treatment_names=None, output_names=None):
+    """
+    Transpose a dictionary of dictionaries, bringing the keys from the second level
+    to the top and vice versa
+
+    Parameters
+    ----------
+    cme_models: function
+        const_marginal_effect function.
+    X: (m, d_x) matrix
+        Features for each sample. Should be in the same shape of fitted X in final stage.
+    d_y: scalar
+        number of outcome.
+    d_t: scalar
+        number of treatment (exclude control in discrete treatment scenario).
+    feature_names: optional None or list of strings of length X.shape[1] (Default=None)
+        The names of input features.
+    treatment_names: optional None or list (Default=None)
+        The name of treatment. In discrete treatment scenario, the name should not include control name.
+    output_names:  optional None or list (Default=None)
+        The name of the outcome.
+
+    Returns
+    -------
+    shap_outs: nested dictionary of Explanation object
+        A nested dictionary by using each Y and each T as a key and the shap_values explanation object as the value.
+
+    """
+    # define masker by using entire dataset, otherwise Explainer will only sample 100 obs by default.
+    background = shap.maskers.Independent(X, max_samples=X.shape[0])
+    shap_outs = defaultdict(dict)
+    for i in range(d_y):
+        for j in range(d_t):
+            def cmd_func(X):
```

Comment:
It would be great if we could call shap fewer times!! 
For instance I expect that d_t would be > 1 more often than d_y.

Can shap explain a multi-output function?

If yes then we could be passing the whole vector of cme's for all t's at once and then just call shap for each d_y separately. 

This way a lot of shap compute will be re-used and it will be much faster.

Reply:
Yes, it does support multi-output function, now I change to what you recommend! Thanks!